src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -409,6 +422,7 @@ class RegularSync(
   private def handleBlockBodies(peer: Peer, m: Seq[BlockBody]): Unit = {
     if (m.nonEmpty) {
       assert(headersQueue.nonEmpty, "handling block bodies while not having any queued headers")
+      assert(importingBlocks, "handling bodies while importing other ones")
```

Comment:
the problem is  that if one of this assertions won't hold then mantis would fail silently later in some really bad state. So i think it is better to keep them, as this can really help with debugging.